### PR TITLE
Refactor Default Config

### PR DESCRIPTION
**Pull request by amir-blox**
Resolve https://github.com/bloxapp/ssv/issues/131

* extract eth1 config to avoid duplication
* provide default values from the relevant package 
  * [contract address](https://github.com/bloxapp/ssv/pull/162/files#diff-0efc389b2e57298ca65e1483fb06e8b58abdfa267c7a885a1083dfc72f74327cR14)
  * [contract ABI](https://github.com/bloxapp/ssv/pull/162/files#diff-511693c8958917db34f43638aee6f06b86f916d48a8701e11cc12ec4606adc43R18)
* Remove old default config ( [./shared/params/](https://github.com/bloxapp/ssv/tree/stage/shared/params) )

### CODE REVIEW

File: eth1/client.go [hunk 1]
```diff
@@ -7,6 +7,13 @@ import (
 	"math/big"
 )
 
+// Options configurations related to eth1
+type Options struct {
+	ETH1Addr          string `yaml:"ETH1Addr" env:"ETH_1_ADDR" env-required:"true"`
+	ETH1SyncOffset    string `yaml:"ETH1SyncOffset" env:"ETH_1_SYNC_OFFSET"`
+	SmartContractAddr string `yaml:"SmartContractAddr" env:"SMART_CONTRACT_ADDR_KEY" env-default:"0x9573C41F0Ed8B72f3bD6A9bA6E3e15426A0aa65B" env-description:"smart contract address"`
```
Comment (alonmuroch) [inline eth1/client.go:14]: I would give it a more precise name like registry contract or something

File: eth1/events.go [hunk 2]
```diff
@@ -14,6 +13,11 @@ import (
...
...bytes"}],"name":"addOperator","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"_ownerAddress","type":"address"},{"internalType":"bytes","name":"_publicKey","type":"bytes"},{"internalType":"bytes[]","name":"_operatorPublicKeys","type":"bytes[]"},{"internalType":"bytes[]","name":"_sharesPublicKeys","type":"bytes[]"},{"internalType":"bytes[]","name":"_encryptedKeys","type":"bytes[]"}],"name":"addValidator","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[],"name":"operatorCount","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"bytes","name":"","type":"bytes"}],"name":"operators","outputs":[{"internalType":"string","name":"name","type":"string"},{"internalType":"address","name":"ownerAddress","type":"address"},{"internalType":"bytes","name":"publicKey","type":"bytes"},{"internalType":"uint256","name":"score","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"validatorCount","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"}]`
```
Comment (alonmuroch) [inline eth1/events.go:18]: this should probably be in a file in the defaults folder/ possible to override via cli options

**Review by alonmuroch — CHANGES_REQUESTED** [review]

**Comment by alonmuroch** [comment]
@amir-blox is it just for the exporter or did you move all configs?

File: eth1/events.go (hunk 2 above)
Comment (amir-blox) [reply eth1/events.go:18]: We currently don't have a central location for defaults, each package defines its config and default values (e.g. [eth1 config](https://github.com/bloxapp/ssv/pull/162/files#diff-0efc389b2e57298ca65e1483fb06e8b58abdfa267c7a885a1083dfc72f74327cR14)), therefore `./eth1` could be a good location.. What about `./eth1/registry_contract/abi.json` ?

File: exporter/node.go [hunk 3]
```diff
@@ -207,7 +207,7 @@ func (exp *exporter) handleValidatorAddedEvent(event eth1.ValidatorAddedEvent) e
 	logger := exp.logger.With(zap.String("pubKey", pubKeyHex))
 	logger.Info("validator added event")
 	// save the share to be able to reuse IBFT functionality
-	validatorShare, err := validator.ShareFromValidatorAddedEvent(event)
+	validatorShare, err := validator.ShareFromValidatorAddedEvent(event, "")
```
Comment (nivBlox) [inline exporter/node.go:210]: how comes that we dont pass operator key here?

File: operator/storage_test.go [hunk 4]
```diff
@@ -98,9 +97,6 @@ func TestSetupPrivateKey(t *testing.T) {
 			sk, err := operatorStorage.GetPrivateKey()
 			require.NoError(t, err)
 
-			if test.existKey == "" && test.passedKey == "" { // new key generated
```
Comment (nivBlox) [inline operator/storage_test.go:101]: do we moved it to differ placE?

**Review by nivBlox — CHANGES_REQUESTED** [review]

File: exporter/node.go (hunk 3 above)
Comment (amir-blox) [reply exporter/node.go:210]: this is only from exporter where there is no operator key

File: operator/storage_test.go (hunk 4 above)
Comment (amir-blox) [reply operator/storage_test.go:101]: `params.SsvConfig().OperatorPublicKey` is no longer exist and therefore this check was removed, before my change the value was populated in `SetupPrivateKey`

File: eth1/events.go (hunk 2 above)
Comment (amir-blox) [reply eth1/events.go:18]: @alonmuroch It is an issue to have a relative filepath as default value (`./eth1/registry_contract/abi.json`), as it needs to be adjusted depends on run location, e.g. when running tests.

I can still enable to load a JSON but let's keep the default hardcoded.

**Review by nivBlox — APPROVED** [review]

File: validator/controller.go [hunk 5]
```diff
@@ -53,6 +55,8 @@ type controller struct {
 
 	validatorsMap       map[string]*Validator
 	newValidatorSubject pubsub.Subject
+
+	operatorPrivateKeyProvider eth1.OperatorPrivateKeyProvider
```
Comment (alonmuroch) [inline validator/controller.go:59]: @amir-blox could be a bit confusing with the operator admin key.
Maybe call it ShareEncryptionKey..?

**Review by alonmuroch — CHANGES_REQUESTED** [review]

**Pull request merged**